hre/src/main/java/hre/config/Configuration.java

```diff
@@ -297,6 +303,53 @@ public static MessageProcessEnvironment getSilicon() throws IOException {
         return env;
     }
 
+    public static String getMonoVersion() {
+        MessageProcessEnvironment env = new MessageProcessEnvironment("mono");
+        env.addArg("--version");
+        MessageProcess mp = env.startProcess();
+        while (!mp.isFinished()) {
+            Output("Waiting for termination of \"mono --version\"...");
+            // Wait for termination or timeout
+        }
+        List<Message> messages = mp.recvAll();
+        Pattern pattern = Pattern.compile("version (\\d+(\\.\\d+)*)", Pattern.CASE_INSENSITIVE);
+        for (Message m : messages) {
+            Matcher matcher = pattern.matcher(m.getFormattedMessage());
+            if (matcher.find()) {
+                return matcher.group(1);
+            }
+        }
+
+        return null;
+    }
+
+    public static void checkCarbonRequirements() {
+        String os = System.getProperty("os.name").toLowerCase();
+        if (os.contains("mac") || os.contains("nix") || os.contains("linux")) {
+            // Mono must exist for boogie to run
+            if (!Paths.commandExists("mono")) {
+                Fail("Unable to detect mono. Please ensure the command \"mono\" is available in the PATH.");
```

Comment:
I think this is a case where "try and ask forgiveness" is better: i.e. just try to obtain the mono version, and output that it's not found if it doesn't work, rather than try to parse `PATH`

Reply:
Good idea